### PR TITLE
feat: use here maps js npm package

### PR DESCRIPTION
**Pull request by KhaledSakr**
BREAKING CHANGE: this also changes the engine when using vector tiles from WEBGL to HARP. Additionally, if a client relied on @types/heremaps, they will encounter some type incompatibilities and might needs to adapt a few things. Some other props were also removed because they no longer make since now that the SDK is bundled, like `onScriptLoadError` and `secure`.

## What the PR Includes
- [x] Use HERE Maps JS bundles as an npm package instead of loading the scripts via loadjs.
- [x] Made a few improvements to types and removed multiple @ts-ignores.

Closes POS-2016

## Checklist
- [ ] Tests are added.
- [ ] Manual testing instructions are added.
- [ ] Configs are added/adapted if applicable.
- [x] Issue is properly linked if applicable.

## Manual Testing Instructions
1. Needs to be tested extensively with the CargoApps. For reference check PR https://github.com/impargo/impargo-apps/pull/9572

### CODE REVIEW

File: src/HEREMap.tsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
+import './bundle'
+
 import { debounce, uniqueId } from 'lodash'
 import React, { forwardRef, useEffect, useImperativeHandle, useRef, useState } from 'react'
 
 import { HEREMapContext } from './context'
+import type { DefaultLayers } from './types'
```
Comment (mowaleed96) [inline src/HEREMap.tsx:7]: it's interesting to know about this syntax 😄

File: src/HEREMap.tsx [hunk 2]
```diff
@@ -18,6 +20,9 @@ export interface HEREMapProps extends H.Map.Options {
   hidpi?: boolean,
   interactive?: boolean,
   lg?: Language,
+  /**
+   * @deprecated no longer supported. HERE Maps JS is now bundled with this package.
+   */
```
Comment (mowaleed96) [inline src/HEREMap.tsx]: As this is a breaking change, should we remove all deprecated options?

File: testbench/index.tsx [hunk 3]
```diff
@@ -11,14 +11,14 @@ if (!process.env.HERE_APIKEY) {
...
   <HEREMap
     apiKey={process.env.HERE_APIKEY}
-    secure
     center={{ lat: 51.528098, lng: 9.951222 }}
     zoom={6}
     truckRestrictions
-    trafficLayer
-    useVectorTiles
+    // trafficLayer
+    // useVectorTiles
```
Comment (mowaleed96) [inline testbench/index.tsx]: I think these were forgotten

File: src/HEREMap.tsx (hunk 2 above)
Comment (KhaledSakr) [reply src/HEREMap.tsx]: That makes sense. At first I was hoping not to make it a breaking change, but I had to change the default engine which has to be a breaking change.

**Review by mowaleed96 — APPROVED** [review]

**Comment by david-paulus-saad** [comment]
RUN URL: https://github.com/impargo/react-here-maps/actions/runs/12469461163
PACKAGE VERSION: 3.0.0-pre-16

**Pull request merged**